contracts/libraries/Ticks.sol

```diff
@@ -120,7 +120,7 @@ library Ticks {
                 require(old.liquidity != 0 && oldNextTick > upper && upperOld < upper, "UPPER_ORDER");
             }
 
-            if (upper <= nearestTick) {
+            if (upper <= TickMath.getTickAtSqrtRatio(currentPrice)) {
```

Comment:
#### update conditions

from `nearestTick` to `poolTick`

* reference : https://github.com/code-423n4/2021-09-sushitrident-2-findings/issues/13